### PR TITLE
Modified sports module for correctness and to contain more precise data

### PR DESCRIPTION
**Pull request by FishOfPitt116**
## Changes
- Updated model of next game methods to return a dictionary to contain more specific information on opponent and venue, as well as if the game is in progress or has completed
- Modified sports documentation to reflect these changes

## Tests
- Wrote more sophisticated unit tests to ensure the structure of each API response is as expected in several cases (most notably, offseason vs in-season)
- Ensured sports unit tests pass

### CODE REVIEW

File: docs/api-reference/SPORTS.md [hunk 1]
```diff
@@ -26,14 +26,35 @@ Returns a `string` containing the current record of the Men's basketball team.
...
+Returns a `dictionary` containing the time and date of the game in UTC, the opponent's ESPN API ID, school name and full team name, whether Pitt is the home or away team, and information on its location. It also returns a 'Status' attribute, which identifies if the game has completed but data in the API has not updated yet.
 #### **Example**:
 
 ###### **Sample Usage**:
 ```python
 >>> next_game_details = get_next_mens_basketball_game()
 >>> print(next_game_details)
-{'full_name': 'Pittsburgh Panthers at Syracuse Orange', 'short_name': 'PITT @SYR', 'season_name': 'Regular Season', 'week': 'Week 19', 'court': 'Spectrum Center'}
+{
+    'Timestamp': '2022-03-08T19:00Z',
```
Comment (RitwikGupta) [inline docs/api-reference/SPORTS.md]: can we maintain the standard of underscore separated key names?

File: docs/api-reference/SPORTS.md [hunk 2]
```diff
@@ -78,15 +99,35 @@ Returns a `string` containing the current record of the football team.
...
 #### **Example**:
 
 ###### **Sample Usage**:
 ```python
 >>> next_game_details = get_next_football_game()
 >>> print(next_game_details)
-{'full_name': 'Pittsburgh Panthers at Penn State Nittany Lions', 'short_name': 'PITT @ PSU', 'season_name': 'Regular Season', 'week': 'Week 3', 'field': 'Beaver Stadium'}
-
+{
+    'Timestamp': '2021-12-05T01:00Z',
```
Comment (RitwikGupta) [inline docs/api-reference/SPORTS.md]: Underscore key names

File: pittapi/sports.py [hunk 3]
```diff
@@ -26,14 +26,6 @@
...
 
-class NextBasketballGame(NamedTuple):
-    name: str
-    short_name: str
-    season_name: str
-    week: int
-    court: str
-
-
 class NextFootballGame(NamedTuple):
```
Comment (RitwikGupta) [inline pittapi/sports.py]: Is this NamedTuple still used? If not, we should delete this class.

File: pittapi/sports.py [hunk 4]
```diff
@@ -56,25 +48,44 @@ def get_mens_basketball_record() -> str:
...
+        elif next_game["competitions"][0]["status"]["type"]["name"] == "STATUS_IN_PROGRESS":
+            status = "IN_PROGRESS"
+        if next_game["competitions"][0]["competitors"][0]["id"] == 221:
+            opponent = next_game["competitions"][0]["competitors"][0]
+            homeaway = next_game["competitions"][0]["competitors"][0]["homeAway"]
+        else:
+            opponent = next_game["competitions"][0]["competitors"][1]
+            homeaway = next_game["competitions"][0]["competitors"][1]["homeAway"]
+        return {
+            "Timestamp" : next_game["date"],
```
Comment (RitwikGupta) [inline pittapi/sports.py]: underscore separated keys

File: pittapi/sports.py [hunk 5]
```diff
@@ -97,22 +108,40 @@ def get_football_record() -> str:
...
+        elif next_game["competitions"][0]["status"]["type"]["name"] == "STATUS_IN_PROGRESS":
+            status = "IN_PROGRESS"
+        if next_game["competitions"][0]["competitors"][0]["id"] == 221:
+            opponent = next_game["competitions"][0]["competitors"][1]
+            homeaway = next_game["competitions"][0]["competitors"][0]["homeAway"]
+        else:
+            opponent = next_game["competitions"][0]["competitors"][0]
+            homeaway = next_game["competitions"][0]["competitors"][1]["homeAway"]
+        return {
+            "Timestamp" : next_game["date"],
```
Comment (RitwikGupta) [inline pittapi/sports.py]: underscore separated keys

File: tests/sports_test.py [hunk 6]
```diff
@@ -27,7 +26,7 @@ def test_get_next_mens_basketball_game(self):
         self.assertIsInstance(sports.get_next_mens_basketball_game(), dict)
 
     def test_get_mens_basketball_standings(self):
```
Comment (RitwikGupta) [inline tests/sports_test.py]: These unit tests where we just assert the type of return data type are weak. Can we instead match against a mock schema instead?

**Review by RitwikGupta — CHANGES_REQUESTED** [review]

**Comment by FishOfPitt116** [comment]
What is the requirement on # of reviewer approvals necessary to merge a pr? Or will you just merge the pr when you find its completed @RitwikGupta

File: pittapi/sports.py (hunk 5 above)
Comment (FishOfPitt116) [reply pittapi/sports.py]: Resolved in [afdce7b](https://github.com/pittcsc/PittAPI/pull/137/commits/afdce7b22c50d0c6f18af707f379a39adcacdd4a)

File: pittapi/sports.py (hunk 4 above)
Comment (FishOfPitt116) [reply pittapi/sports.py]: Resolved in [afdce7b](https://github.com/pittcsc/PittAPI/pull/137/commits/afdce7b22c50d0c6f18af707f379a39adcacdd4a)

File: pittapi/sports.py (hunk 3 above)
Comment (FishOfPitt116) [reply pittapi/sports.py]: Resolved in [afdce7b](https://github.com/pittcsc/PittAPI/pull/137/commits/afdce7b22c50d0c6f18af707f379a39adcacdd4a)

File: docs/api-reference/SPORTS.md (hunk 2 above)
Comment (FishOfPitt116) [reply docs/api-reference/SPORTS.md]: Resolved in [afdce7b](https://github.com/pittcsc/PittAPI/pull/137/commits/afdce7b22c50d0c6f18af707f379a39adcacdd4a)

File: docs/api-reference/SPORTS.md (hunk 1 above)
Comment (FishOfPitt116) [reply docs/api-reference/SPORTS.md]: Resolved in [afdce7b](https://github.com/pittcsc/PittAPI/pull/137/commits/afdce7b22c50d0c6f18af707f379a39adcacdd4a)

File: tests/sports_test.py (hunk 6 above)
Comment (FishOfPitt116) [reply tests/sports_test.py]: Resolved in [3258f4b](https://github.com/pittcsc/PittAPI/pull/137/commits/3258f4b0e34c25565c99abf67ab9f9716c6fad5c)

File: pittapi/sports.py [hunk 7]
```diff
@@ -56,37 +36,56 @@ def get_mens_basketball_record() -> str:
...
+                "name" : opponent["team"]["displayName"]
+            },
+            "home_away" : homeaway,
+            "location" : {
+                "full_name" : next_game["competitions"][0]["venue"]["fullName"],
+                "address" : next_game["competitions"][0]["venue"]["address"]
+            },
+            "status" : status
+        }
+    except IndexError:
```
Comment (RitwikGupta) [inline pittapi/sports.py:72]: Is offseason the only time an `IndexError` can occur?

Comment (RitwikGupta) [reply pittapi/sports.py:72]: It would be better if there was a more robust way to check if it's offseason.

**Review by RitwikGupta — COMMENTED** [review]
Looks good to me. Once you address the offseason question in the positive or negative, I can merge.

File: pittapi/sports.py (hunk 7 above)
Comment (FishOfPitt116) [reply pittapi/sports.py:72]: Yeah... the IndexError can only occur when there isn't a next game, therefore implying it's the offseason. I'll add a comment specifying this.

I ran a couple of sample requests on the ESPN API with teams whose last game had already been played (most notably Pitt basketball), and the nextEvent field in the response was their last game of the previous season. In that case, our API will return a response with the game info but status "GAME_COMPLETE," which seems to be more common for the offseason. 

All this to say the IndexError simply covers an edge case that could potentially occur.

File: pittapi/sports.py [hunk 8]
```diff
@@ -129,6 +130,7 @@ def get_next_football_game() -> dict:
             "status" : status
         }
     except IndexError:
+        # IndexError occurs when a next game on the schedule is not present, implying it's the offseason.
```
Comment (RitwikGupta) [inline pittapi/sports.py]: Let' just return `status: NO_GAME_SCHEDULED` or something like it. There may be other reasons no future game is scheduled -- strikes, NCAA imposed penalties, etc. Instead if attempting to interpret all possible causes, let's just be very exact and say exactly why an `IndexError` may occur, i.e., there is no next game.

**Review by RitwikGupta — CHANGES_REQUESTED** [review]

**Review by RitwikGupta — APPROVED** [review]

**Pull request merged**